x-pack/packages/kbn-elastic-assistant/impl/connectorland/connector_setup/index.test.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useConnectorSetup } from '.';
+import { act, renderHook } from '@testing-library/react-hooks';
+import { fireEvent, render } from '@testing-library/react';
+import { alertConvo, welcomeConvo } from '../../mock/conversation';
+import { TestProviders } from '../../mock/test_providers/test_providers';
+import { EuiCommentList } from '@elastic/eui';
+
+const onSetupComplete = jest.fn();
+const defaultProps = {
+  conversation: welcomeConvo,
+  onSetupComplete,
+};
+const newConnector = { actionTypeId: '.gen-ai', name: 'cool name' };
+jest.mock('../add_connector_modal', () => ({
+  // @ts-ignore
+  AddConnectorModal: ({ onSaveConnector }) => (
+    <>
+      <button
+        type="button"
+        data-test-subj="modal-mock"
+        onClick={() => onSaveConnector(newConnector)}
+      />
+    </>
+  ),
+}));
+
+const setConversation = jest.fn();
+const setApiConfig = jest.fn();
+const mockConversation = {
+  appendMessage: jest.fn(),
+  appendReplacements: jest.fn(),
+  clearConversation: jest.fn(),
+  createConversation: jest.fn(),
+  deleteConversation: jest.fn(),
+  setApiConfig,
+  setConversation,
+};
+
+jest.mock('../../assistant/use_conversation', () => ({
+  useConversation: () => mockConversation,
+}));
+
+jest.spyOn(global, 'clearTimeout');
+describe('useConnectorSetup', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('should render comments and prompts', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook(() => useConnectorSetup(defaultProps), {
+        wrapper: ({ children }) => (
+          <TestProviders
+            providerContext={{
+              getInitialConversations: () => ({
+                [alertConvo.id]: alertConvo,
+                [welcomeConvo.id]: welcomeConvo,
+              }),
+            }}
+          >
+            {children}
+          </TestProviders>
+        ),
+      });
+      await waitForNextUpdate();
+      expect(
+        result.current.comments.map((c) => ({ username: c.username, timestamp: c.timestamp }))
+      ).toEqual([
+        {
+          username: 'You',
+          timestamp: 'at: 7/17/2023, 1:00:36 PM',
+        },
+        {
+          username: 'Elastic AI Assistant',
+          timestamp: 'at: 7/17/2023, 1:00:40 PM',
+        },
+      ]);
+
+      expect(result.current.prompt.props['data-test-subj']).toEqual('prompt');
+    });
+  });
+  it('should set api config for each conversation when new connector is saved', async () => {
```

Comment:
This is a good one to have -- ty! <img width="16" src="https://user-images.githubusercontent.com/2946766/141219243-86619f7f-c300-4355-96f5-316d70aa6a0e.png" /> 

Should only be the case if there were no connectors previously. Intent being, all existing conversations will 'heal' when first working connector is added, so the user doesn't have to manually set the connector on all existing conversations.